framework/src/fviks/FVInterfaceKernel.C

```diff
@@ -188,8 +188,11 @@ FVInterfaceKernel::computeResidual(const FaceInfo & fi)
 
   const auto r = MetaPhysicL::raw_value(fi.faceArea() * fi.faceCoord() * computeQpResidual());
 
-  addResidual(_elem_is_one ? r : -r, _var1.number(), _elem_is_one ? false : true);
-  if (_var1.sys().number() == _var2.sys().number())
+  // If the two variables belong to two different nonlinear systems, we only contribute to the one
+  // which is being assembled right now
+  if (_var1.sys().number() == _subproblem.currentNlSysNum())
```

Comment:
I think we should assert this to be true because we determine the interface kernels to be run by filtering on `AttribSysNum`